### PR TITLE
Experimental ARM64 Ubuntu 22.04 Support (With Warning)

### PR DESCRIPTION
**Pull request by aviwad**
### Description

Enables experimental support for Ubuntu 22.04 on Arm. Especially useful for M1 MacBooks. Setup script delivers a warning if detected to be running on Arm. Depends on packages from Ubuntu Mantic, causing Frankenstein / problems in mixing repositories within Ubuntu.

### Testing Done

Installed on a fresh Ubuntu 22.04 ARM64 VM, and Thunderscope runs perfectly after running the setup script.

### Resolved Issues

- Install ARM64 of GameController for ARM
- Install ARM64 Bazel for ARM
- Don't add lines to .bashrc if they already exist (such as Bazel autocompletion)
- add aarch64/arm64 compilation toolchain support

### Changes
- Use Qt5 installation symlinked in /opt/tbotspython/qt5/ to be architecture-agnostic
- If on arm: compiles PyQt6 from source
- If on arm: installs PyQt6 & PyQt6-WebEngine via ubuntu mantic repository (breaks ubuntu installation, frankenstein-ubuntu)
- If on arm: enables tbotspython to use global python dist_packages to access distribution's PyQt6 (adds environment variables PYTHONPATH=/usr/lib/python3/dist-packages/  and PYQTGRAPH_QT_LIB=PyQt6  to bashrc

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [-] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [-] **Remove all commented out code**
- [-] **Remove extra print statements**: for example, those just used for testing
- [-] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue

### CODE REVIEW

File: environment_setup/setup_software.sh [hunk 1]
```diff
@@ -126,21 +155,78 @@ if [[ $(lsb_release -rs) == "22.04" ]]; then
...
+    sudo rm /etc/apt/sources.list.d/temp.list
+    # allow tbotspython to access dist-packages
+    add_bashrc_if_not_there "export PYTHONPATH=/usr/lib/python3/dist-packages/"
+    # fix for pyqtgraph trying to use pyqt5 by default
+    add_bashrc_if_not_there "export PYQTGRAPH_QT_LIB=PyQt6"
+    
+    # change default qt to qt5 again
+else
+    # else if x86_64, install PyQt6 normally using pip
+    sudo /opt/tbotspython/bin/pip3 install pyqt6==6.5.0 PyQt6-WebEngine
```
Comment (nimazareian) [inline environment_setup/setup_software.sh]: We should fix version of `PyQt6-WebEngine`
```suggestion
    sudo /opt/tbotspython/bin/pip3 install pyqt6==6.5.0 PyQt6-WebEngine==6.5.0
```

File: src/software/thunderscope/requirements.txt [hunk 2]
```diff
@@ -1 +1,2 @@
 pyqtgraph==0.13.3
+numpy==1.24.4
```
Comment (nimazareian) [inline src/software/thunderscope/requirements.txt:2]: This is already in `ubuntu20_requirements.txt` and `ubuntu22_requirements.txt`, why do we need it here?

**Review by nimazareian — COMMENTED** [review]
I had some initial thoughts

We should consider adding an `aarch64` linux to our CI, similar to how #3001 adds one for Ubuntu 22.04. Since it's currently impossible for anyone without an M1 to tell if the current or future changes are gonna break arm support.

File: src/software/thunderscope/requirements.txt (hunk 2 above)
Comment (itsarune) [reply src/software/thunderscope/requirements.txt:2]: Bazel sandboxes each build I think, so when you set up thunderscope_main, it won't have numpy until you set it as a requirement

Comment (aviwad) [reply src/software/thunderscope/requirements.txt:2]: I believe that was added by @AdenC123.

Comment (AdenC123) [reply src/software/thunderscope/requirements.txt:2]: As arun said- I was unable to run thunderscope_main directly without fixing the numpy version here

**Comment by aviwad** [comment]
> I had some initial thoughts
> 
> We should consider adding an `aarch64` linux to our CI, similar to how #3001 adds one for Ubuntu 22.04. Since it's currently impossible for anyone without an M1 to tell if the current or future changes are gonna break arm support.

https://github.com/actions/runner-images/issues/5631

Don't think it's possible right now.

**Comment by aviwad** [comment]
@itsarune adding --cpu=k8 didn't fix the CI unfortunately. @AdenC123 could you take a look if possible?

**Comment by nimazareian** [comment]
Given that progress has been halted on this PR and Arm support, I think we should close this PR. What do you think? @aviwad @itsarune

**Comment by aviwad** [comment]
I think so. Arm support is only realistic once we target Ubuntu 24.04 and Python 3.12+.

**Pull request closed without merging**